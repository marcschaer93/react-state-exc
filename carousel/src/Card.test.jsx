import { render } from "@testing-library/react";
import Card from "./Card";
import { expect } from "vitest";

// Mock Card Data
const testCard = [
  { caption: "Foto with a test 1", src: "testFoto1" },
  { caption: "Foto with a test 2", src: "testFoto2" },
];
// smoke Test
it("should render", () => {
  <Card
    caption={testCard.caption}
    src={testCard.src}
    currNum={0 + 1}
    totalNum={2}
  />;
});

// snapshot Test
it("should match snapshot", () => {
  const { asFragment } = render(
    <Card
      caption={testCard.caption}
      src={testCard.src}
      currNum={0 + 1}
      totalNum={2}
    />
  );
  expect(asFragment()).toMatchSnapshot();
});
