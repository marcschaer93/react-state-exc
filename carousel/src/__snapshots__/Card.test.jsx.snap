// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="Card"
  >
    <h4
      class="Card-title"
    />
    <img
      class="Card-image"
    />
    <small
      class="Card-small"
    >
      Image 1 of 2.
    </small>
  </div>
</DocumentFragment>
`;
